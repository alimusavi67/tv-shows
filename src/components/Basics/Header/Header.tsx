import React from 'react';
import TimeZoneSelector from '../../TimeZoneSelector/TimeZoneSelector';

const Header: React.FC = () => {
  return (
    <header className="p-4 bg-gray-800 text-white flex flex-col md:flex-row justify-between items-center">
      <div className="flex items-center text-2xl font-bold">
        <a href="/">TV Shows DB</a>
      </div>
      <TimeZoneSelector />
    </header>
  );
};

export default Header;
