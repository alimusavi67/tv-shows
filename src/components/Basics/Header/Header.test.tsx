import React from 'react';
import { render } from '@testing-library/react';
import Header from './Header';

jest.mock('../../TimeZoneSelector/TimeZoneSelector', () => {
  return function MockTimeZoneSelector() {
    return <div>TimeZoneSelector Mock</div>;
  };
});

describe('Header component', () => {
  it('matches the snapshot', () => {
    const { asFragment } = render(<Header />);
    expect(asFragment()).toMatchSnapshot();
  });
});
