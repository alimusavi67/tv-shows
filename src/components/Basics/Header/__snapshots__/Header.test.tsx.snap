// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component matches the snapshot 1`] = `
<DocumentFragment>
  <header
    class="p-4 bg-gray-800 text-white flex flex-col md:flex-row justify-between items-center"
  >
    <div
      class="flex items-center text-2xl font-bold"
    >
      <a
        href="/"
      >
        TV Shows DB
      </a>
    </div>
    <div>
      TimeZoneSelector Mock
    </div>
  </header>
</DocumentFragment>
`;
